app/facades/facet_configuration_facade.rb

```diff
@@ -27,10 +27,20 @@ def all_solr_fields
 
   def solr_field_iterator
     @solr_field_iterator ||= iterator_factory.new(all_solr_fields).map do |k, v|
+      setup_facet_item_set(v) if k == :set
       [k, HasFacetField.new(v, grouped_facets[k].try(:first) || FacetField.new(:key => k))]
     end
   end
 
+  def setup_facet_item_set(value)
+    top_terms = value.topTerms.each_slice(2).to_a
```

Comment:
I'm making a guess here, so I could be way off, but it looks like you could never configure items that aren't in the top terms list.  Is that what we really want?  (I'm not actually finding what it is that `topTerms` means)